Add spec text

jridgewell opened the pull request.

jridgewell commented:
Merging so that the spec text can be viewed. Please still add reviews, and I'll address in follow ups.

ljharb commented on spec.emu:
```diff
@@ -4,20 +4,308 @@
+      <p>When removing the common leading indentation, lines which contain only whitespace will have all whitespace removed.</p>
+    </emu-note>
+    <emu-alg>
+      1. If Type(_templateOrFn_) is Object and IsCallable(_templateOrFn_) is *true*, then
```
Can the type not be object, while IsCallable is true?

jridgewell replied:
Actually, `IsCallable` checks type is `Object`. I'll remove.

jridgewell replied:
af8443f

ljharb commented:
The spec also always throws in the callable branch - 1.b. makes an Abstract Closure and does nothing with it.

ljharb commented:
Also, LeadingQhitespaceString references an S that doesn’t exist - and I’m not sure the prose is clear enough to explain what the AO should be doing. It’d probably be better to do something more explicit.

jridgewell commented:
> The spec also always throws in the callable branch - 1.b. makes an Abstract Closure and does nothing with it.

Fixed locally. I have a large refactor coming to split the huge `DedentStringsArray` into smaller AOs.

> Also, LeadingWhitespaceSubstring references an S that doesn’t exist

Fixed.



> and I’m not sure the prose is clear enough to explain what the AO should be doing. It’d probably be better to do something more explicit.

I took this wording from [TrimString](https://tc39.es/ecma262/multipage/text-processing.html#sec-trimstring). I'm open to anything that lets me say "if this is Unicode white space".

> IsAllWhitespace’s name says to me that it’s true when the string contains nothing but whitespace

That's the intention.

jridgewell commented:
Pushed AOs and fixes in [f9d3621](https://github.com/tc39/proposal-string-dedent/commit/f9d36218db56838da4889c1e6ba88d7a41385a6f).

ljharb commented:
If that's the intention, can it just _call_ TrimString and assert that it receives an empty string back?
